cmd/vulcan-trivy/vulcan-trivy.go

```diff
@@ -0,0 +1,337 @@
+		triviArgs = append(triviArgs, severitiesFlag...)
+	}
+	// Append the target (docker image including registry hostname).
+	triviArgs = append(triviArgs, target)
+
+	log.Printf("Running command: %s %s\n", triviCmd, triviArgs)
+
+	err := retry.Do(
+		func() error {
+			cmd := exec.Command(triviCmd, triviArgs...)
+			cmdOutput, err := cmd.CombinedOutput()
+			if err != nil {
+				log.Printf("exec.Command() failed with %s\nCommand output: %s\n", err, string(cmdOutput))
+				return errors.New("trivy command execution failed")
+			}
+			log.Printf("trivy command execution completed successfully")
+			return nil
+		},
+		retry.Attempts(3),
+		retry.DelayType(retry.RandomDelay),
+		retry.MaxJitter(5*time.Second),
+	)
+	if err != nil {
+		log.Printf("retry exec.Command() failed with error: %s\n", err)
+		return errors.New("trivy command execution failed")
+	}
+
+	jsonFile, err := os.Open(reportOutputFile)
+	if err != nil {
+		log.Printf("trivy report output file read failed with error: %s\n", err)
+		return errors.New("trivy report output file open failed")
+	}
+	log.Printf("successfully open trivy report output file %s\n", reportOutputFile)
+	defer jsonFile.Close()
+
+	byteValue, err := ioutil.ReadAll(jsonFile)
+	if err != nil {
+		log.Printf("trivy report output file read failed with error: %s\n", err)
+		return errors.New("trivy report output file read failed")
+	}
+
+	var results ScanResponse
+	err = json.Unmarshal(byteValue, &results)
+	if err != nil {
+		return errors.New("unmarshal trivy output failed")
+	}
+
+	// If there are no vulnerabilities we can return.
+	if len(results) < 1 {
+		return nil
+	}
+
+	ap := report.ResourcesGroup{
+		Name: "Affected Packages",
+		Header: []string{
+			"Name",
+			"Version",
+			"Severity",
+			"FixedBy",
+		},
+	}
+
+	vp := report.ResourcesGroup{
+		Name: "Package Vulnerabilities",
+		Header: []string{
+			"Name",
+			"Version",
+			"Vulnerabilities",
+		},
+	}
+
+	var rows []map[string]string
+	duppedPackageVulns := make(map[string]map[string]string)
+	// As we are scanning only one container per check, we have only one item
+	// in results array.
+	for _, feature := range results[0].Vulnerabilities {
+		// Set global score for the report.
+		score := getScore(feature.Severity)
+		if score > vuln.Score {
+			vuln.Score = score
+		}
+
+		ap := map[string]string{
+			"Name":            feature.PkgName,
+			"Version":         feature.InstalledVersion,
+			"Severity":        feature.Severity,
+			"FixedBy":         feature.FixedVersion,
+			"Vulnerabilities": fmt.Sprintf("%s", feature.Description),
+		}
+		if len(ap["Vulnerabilities"]) > 512 {
+			reportTruncated = true
+			ap["Vulnerabilities"] = fmt.Sprintf("%s ...truncated", ap["Vulnerabilities"][0:500])
+		}
+
+		// Check if affected package has already been indexed.
+		if _, exists := duppedPackageVulns[ap["Name"]]; !exists {
+			duppedPackageVulns[ap["Name"]] = ap
+			continue
+		}
+
+		key := duppedPackageVulns[ap["Name"]]
```

Comment:
This access is being done some lines above. Maybe you can rewrite the while thing as something like:
```
elem, ok := duppedPackageVulns[ap["Name"]]
if !ok {
...
}
```

Reply:
Good catch. Fixed.